politeiawww/user/cockroachdb/cockroachdb_test.go

```diff
@@ -0,0 +1,381 @@
+// +build cockroachdb
+
+// The tests in this file need cockroachdb to be up and will only be run if you
+// specify "-tags=cockroachdb" with `go test`.
+
+package cockroachdb
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/decred/politeia/politeiawww/user"
+	"github.com/google/uuid"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+)
+
+func model2session(s Session) user.Session {
+	return user.Session{
+		ID:        s.ID,
+		UserID:    s.UserID,
+		MaxAge:    s.MaxAge,
+		CreatedAt: s.CreatedAt.Unix(),
+	}
+}
+
+func connect2testdb() (*cockroachdb, error) {
+	home := os.Getenv("HOME")
+	var (
+		tdbhost        string = "localhost:26257"
+		tnetwork       string = "testnet3"
+		tdbrootcert    string = path.Join(home, ".cockroachdb/certs/clients/politeiawww/ca.crt")
+		tdbcert        string = path.Join(home, ".cockroachdb/certs/clients/politeiawww/client.politeiawww.crt")
+		tdbkey         string = path.Join(home, ".cockroachdb/certs/clients/politeiawww/client.politeiawww.key")
+		tencryptionkey string = path.Join(home, ".politeiawww/sbox.key")
+	)
+	return New(tdbhost, tnetwork, tdbrootcert, tdbcert, tdbkey, tencryptionkey)
+}
+
+func TestSessionMethods(t *testing.T) {
```

Comment:
Why run these in a single test instead of having each as its own test?  Doesn't that just make it harder to track down failures?

Also, in general, I prefer named, table driven tests.  Sometimes its not appropriate though.

https://github.com/golang/go/wiki/TableDrivenTests

Reply:
I grouped them all in a single test so that `cockroachdb` is initialized once only. Will look into the able driven test approach.

Reply:
Every check is now a independent test function as you suggested.